Provide easier access to evcc web interface

DerAndereAndi opened the pull request:
- Use mdns to provide access to the evcc website via `http://evcc.local:7070`
- Correctly shutdown the mdns announcement on exit
- Unknown: We need to check how this works in a docker environment, as that may provide unreachable IP addresses

andig commented on server/http.go:
```diff
@@ -114,3 +119,17 @@ func NewHTTPd(url string, site site.API, hub *SocketHub, cache *util.Cache) *HTT
 	return s.Handler.(*mux.Router)
 }
+
+func (s *HTTPd) Announce() (*zeroconf.Server, error) {
```
Wir brauchen nur den Port, oder? Dann sollte das announcen eine eigene Komponente werden. z.B. `server.AnnounceMDNS`. Hat ja mit httpd nix zu tun.

DerAndereAndi replied:
Habs eingebaut. Schau es dir mal bei Gelegenheit an.

andig commented:
Müsste mit Docker getestet werden. Tut vmtl. nicht, sollte aber auch nicht schaden.

DerAndereAndi committed:
Provide easier access to evcc web interface

- Use mdns to provide access to the evcc website via `http://evcc.local:7070`
- Correctly shutdown the mdns announcement on exit
- Unknown: We need to check how this works in a docker environment, as that may provide unreachable IP addresses

DerAndereAndi committed:
Move AnnounceMDNS to server

andig commented on cmd/root.go:
```diff
@@ -190,6 +190,18 @@ func run(cmd *cobra.Command, args []string) {
 	socketHub := server.NewSocketHub()
 	httpd := server.NewHTTPd(uri, site, socketHub, cache)
 
+	// announce webserver on mDNS
```
Dir url kennen wir doch- warum soll das über den httpd gehen? Die Abhängigkeit ist egtl nicht nötig.

DerAndereAndi committed:
Simplify

andig commented on server/mdns.go:
```diff
@@ -0,0 +1,8 @@
+
+// Announce evcc on a given HTTP port
+func AnnounceMDNS(instance, service, host string, port int) (*zeroconf.Server, error) {
+	return zeroconf.RegisterProxy("evcc Website", "_http._tcp", "local.", port, "evcc", nil, []string{}, nil)
```
Da sollte der linter aber meckern. Fast alle Parameter unbenutzt 😳

DerAndereAndi replied:
ups

andig commented on cmd/root.go:
```diff
@@ -190,6 +192,21 @@ func run(cmd *cobra.Command, args []string) {
+				log.ERROR.Printf("failed to announce webserver: %s", err)
+			}
+
+			shutdown.Register(func() {
```
Die func brauchts nicht, einfach das shutdown registrieren

DerAndereAndi replied:
noch ein ups. war ein überbleibsel

andig commented on cmd/root.go:
```diff
@@ -190,6 +192,21 @@ func run(cmd *cobra.Command, args []string) {
+	// announce webserver on mDNS
+	if _, port, err := net.SplitHostPort(uri); err == nil {
+		portInt, err := strconv.Atoi(port)
+		if err == nil {
```
if portint….

DerAndereAndi replied:
joah

andig commented on cmd/root.go:
```diff
@@ -190,6 +192,21 @@ func run(cmd *cobra.Command, args []string) {
+		portInt, err := strconv.Atoi(port)
+		if err == nil {
+			zc, err := server.AnnounceMDNS("evcc Website", "_http._tcp", "evcc", portInt)
+			if err != nil {
```
Da das announce auch nur ein 1zeiler ist kanns auch gleich hier mit rein?

DerAndereAndi replied:
bin mir nicht sicher was du hier meinst. schau es dir bitte einfach mal an was ich geändert hab

DerAndereAndi committed:
Style fixes and actually use fct params

andig committed:
wip
